### PR TITLE
Fix emails from contact.php being rejected by correctly configured mail servers

### LINKED ISSUE #257: Change to SMTP mail composition to comply with standards
> **Problem:**  
> ## When a user writes an Email (PM) to another user, the mail header is constructed in this manner now (only the important lines):  
> 
> Return-Path: <fromemail>  
> To: <toemail>  
> From: <fromemail>  
> ## Reply-To: <fromemail>  
> 
> This (often) results in the mail server sending an error : <ins>_"The mail server does not understand the RCPT TO command. Reason: 553 5.7.1. Sender address rejected: not owned by user XXXXXX."_</ins>  
> As it seems, the attempt to send an email from [user@xxx.com](mailto:user@xxx.com) from the mailserver yyy.com is seen as a potential abuse and is rejected.
> ## **The mail should therefor be constructed (correctly) in this manner:**  
> 
> Return-Path: <Mybb_name> <Mybb_returnmail/adminmail>  
> To: <toname> <toemail>  
> From: <Mybb_name <Mybb_adminmail>  
> ## Reply-To: <fromname> <fromemail>  
> 
> **This scheme should be mandatory for ALL emails sent by the system.**  
> The names in front of the email adresses are required to avoid confusion for mail programs (e.g. mail programs like to use the "from name" when replying to a message when the "reply-to name" is empty (despite of the email).

Original report: http://dev.mybb.com/issues/1720
Reporter: Yuri Taillandier
Status: New

### PR DESCRIPTION
**Pull request by kawaii**
This fixes #257 and another few issues I've seen related to correctly configured SMTP servers rejecting mail from MyBB's contact form due to it trying to spoof an address that the SMTP server has no control over.

### CODE REVIEW

**Comment by euantorano** [comment]
This removes the use for the `email` input on the contact page. I see a few options:

- Remove that input field (meaning that admins might not be able to email the user back
- Add a setting to allow spoofing of from address
- Append the `email` field to the body of the email

Thoughts @mybb/developers @mybb/sqa

**Comment by kawaii** [comment]
Appending the email to the body of the message seems like the sanest solution to me.

File: contact.php [hunk 1]
```diff
@@ -223,7 +223,7 @@
 		$message = $lang->sprintf($lang->email_contact, $mybb->input['email'], $user, $session->ipaddress, $mybb->input['message']);
 
 		// Email the administrator
-		my_mail($contactemail, $subject, $message, $mybb->input['email']);
+                my_mail($contactemail, $subject, $message, $mybb->settings['adminemail']);
```
Comment (Shade-) [inline contact.php]: The indentation seems to be incorrect. We use tabs, not spaces.

**Comment by euantorano** [comment]
Actually the best option should be to set the `Reply-To` mail header using the `$return_email` parameter of `my_mail`: https://github.com/mybb/mybb/blob/3902684d7c974ad4a9ececdd049abdc216ff7f3d/inc/functions.php#L566

**Comment by euantorano** [comment]
Updated PR in #2877 using the `$return_email` parameter.

**Pull request closed without merging**